setup.py

```diff
@@ -44,7 +44,9 @@ def load_readme():
         "coloredlogs",
         "chainmap ; python_version<'3'",
     ],
-    extras_require={},
+    extras_require={"networkx": ["networkx"],
```

Comment:
If I'm understanding, this is `extra_key` that maps to a list of `[extra_requirement1, ...]`. Would it make sense to make this something like:

```
extras_require={"vis": ["networkx", "matplotlib", "pygrapyviz"],
}
```

Reply:
That's a good idea.  Also be good if i fix it so pygraphviz is spelled correctly.

Reply:
It might be good to document in the installation information how to install the `vis` aspects too.

Reply:
I was trying to test this and ended up having to install the requirements manually. Might be worth adding them to the requirements.txt file as pip doesn't seem to allow optional specification when installing from a repository locally.

Reply:
Yeah, and graphviz has to be manual no matter what for now, which is kind of annoying; definitely needs some docs for that.  

So, I'd thought the extras provisions in the setup.py would have taken care of that, preventing the need for multiple requirements files..   That's disappointing to hear it doesn't work out that way..

Reply:
What I meant was that during testing, since I had to install it via `pip install .` it didn't allow me to use the optional flags. I couldn't figure out how to get it to work, and noticed they weren't in the `requirements.txt` -- so those should be there for developers to use. Otherwise, I think it's just compiling a wheel and add that path using `-f` to test is the optionals work I think?